bofire/benchmarks/aspen_benchmark.py

```diff
@@ -0,0 +1,199 @@
+
+        Returns:
+            pd.DataFrame: Output values from Aspen. The dataframe includes valid_(variable_name) columns for
+            each output variable when the simulation went successful.
+        """
+
+        # Only start Aspen, when it is not already blocking.
+        if self.aspen_is_running is False:
+            self.start_aspen()
+
+        # Make inputs Aspen-readable
+        if self.translate_into_aspen_readable is not None:
+            candidates = self.translate_into_aspen_readable(
+                domain=self.domain,  # type: ignore
+                candidates=candidates,
+            )
+
+        y_outputs = {}
+        add_outputs = {}
+        for key in self.output_keys:
+            y_outputs[key] = []
+            y_outputs["valid_" + key] = []
+        for key in self.additional_outputs:
+            add_outputs[key] = []
+        # Iterate through dataframe rows to retrieve multiple input vectors. Running seperate simulations for each.
+        for index, row in candidates.iterrows():
+            logger.info("Writing inputs into Aspen")
+            # Write input variables corresping to columns into aspen according to predefined paths.
+            for key in self.input_keys:
+                try:
+                    aspen.Tree.FindNode(self.paths.get(key)).Value = row[key]
+                except ConnectionAbortedError:
+                    logger.exception("Not able to write " + key + " into Aspen.")
+                    raise ValueError("Not able to write " + key + " into Aspen.")
+
+            # Reset Aspen simulation
+            aspen.Reinit()
+            # Start new Aspen simulation
+            logger.info("Aspen simulation run " + str(index))
+            aspen.Engine.Run2()
+            logger.info("Simulation done.")
+
+            # Retrieve outputs from Aspen and write into data frame
+            logger.info("Retrieving outputs from Aspen.")
+            try:
+                # Check for errors during simulation in Aspen that disqualify the results
+                status = aspen.Tree.FindNode(
+                    "\\Data\\Results Summary\\Run-Status\\Output\\UOSSTAT2"
+                ).Value
+
+                for key in self.output_keys:
+                    y_outputs[key].append(
+                        aspen.Tree.FindNode(self.paths.get(key)).Value
+                    )
+                for key in self.additional_outputs:
+                    add_outputs[key].append(
+                        aspen.Tree.FindNode(self.paths.get(key)).Value
+                    )
+
+                for key in self.domain.outputs.get_keys():
+                    if status == 8:
+                        # Result is valid and add valid_var = 1
+                        # Status = 8 corresponds to a valid result that should be kept, 10 is a warning, 9 does not converge
+                        y_outputs["valid_" + key].append(1)
```

Comment:
```suggestion
                        y_outputs[f"valid_{key}" for key in self.domain.outputs.get_keys()] = 1
```

Reply:
I implemented your suggestion. The integer still needs to be appended to the list within the dictionary. Otherwise, it sets the whole list to 1 or 0 in case the function evaluates multiple input vectors.